Authentication.md

```diff
@@ -199,6 +200,120 @@ service = ExampleServiceV1.new_instance(service_name='example_service')
 ```
 
 
+## Identity and Access Management (IAM) Authentication (grant type: assume)
+The `IAMAssumeAuthenticator` performs a two-step token fetch sequence to obtain
```

Comment:
That is interesting - perhaps capitalizing acronyms is idiomatic in Python?

Reply:
I have to argue here, because I don't think it's unfortunate. :) I think it's more readable, or at least easier to understand that the part of the word is an acronym - especially when you are working with a unfamiliar code base.

---

bad joke: ON
`IamAuthenticator` --> What? I am not an authenticator...
`IAMAuthenticator` --> Ohh, that must be Identity and Access Management, I got it!
bad joke: OFF

---

> That is interesting - perhaps capitalizing acronyms is idiomatic in Python?

It is! From the official PEP styleguide: `Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.` ([source](https://peps.python.org/pep-0008/#descriptive-naming-styles))

This is exactly the same in Go. I thought it's in the "Effective Go" but I was not able to find it there, so here are a few examples from the stdlib instead: [type DB struct](https://pkg.go.dev/database/sql#DB), [ListenAndServerTLS](https://pkg.go.dev/net/http#ListenAndServeTLS), [ParseHTTPVersion](https://pkg.go.dev/net/http#ParseHTTPVersion)

Reply:
ok, so it's the other three languages that are incorrect... my mistake :)

Reply:
Not necessarily. I think it depends on the language and what the official style guide - or the one that we want to follow - says. Sorry if I was too harsh!